### PR TITLE
Add MathInstruct

### PR DESCRIPTION
**Pull request by mmhamdy**
Hi everyone 👋

This PR adds the MathInstruct Dataset (with the downloader, preparer, and mapper). All tests passed successfully.

Tell me what you think, happy to make any changes.

### CODE REVIEW

File: data_summaries/MathInstruct.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            "The Ohio State University"
+        ],
+        "Licenses": [
+            {
+                "License": "MIT License",
+                "License URL": "https://github.com/TIGER-AI-Lab/MAmmoTH#license"
+            }
+        ],
+        "License Notes": "The authors provided a table for licenses of the original datasets. Although one dataset has a cc-by-nc license, the authors chose an MIT license for their curated dataset",
+        "License Verified By": "",
```
Comment (anthonywchen) [inline data_summaries/MathInstruct.json:57]: Add your name here :)

File: data_summaries/MathInstruct.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        ],
+        "Creators": [
+            "University of Waterloo",
+            "The Ohio State University"
+        ],
+        "Licenses": [
+            {
+                "License": "MIT License",
+                "License URL": "https://github.com/TIGER-AI-Lab/MAmmoTH#license"
+            }
```
Comment (anthonywchen) [inline data_summaries/MathInstruct.json:54]: Add OpenAI License here in addition to the GitHub license.

**Review by anthonywchen — APPROVED** [review]
Looks good! Just address to minor points and the merge conflict and it's ready to go

File: data_summaries/MathInstruct.json (hunk 1 above)
Comment (mmhamdy) [reply data_summaries/MathInstruct.json]: Done

File: data_summaries/MathInstruct.json (hunk 2 above)
Comment (mmhamdy) [reply data_summaries/MathInstruct.json:58]: Done

**Comment by anthonywchen** [comment]
Thanks for the PR! merged.

**Pull request merged**